### ISSUE TITLE
[Feature] Add an option in SmbShare to expose the share as a Service

### ISSUE THREAD

**Issue by nixpanic** [samba-in-kubernetes/samba-operator#59]
For users it will be very convenient to have a [Service](https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/) that can be use to connect to the SmbShare by name.

For example, using `tests/files/smbshare1.yaml` as SmbShare, and adding the following Service:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: tshare1
  namespace: samba-operator-system
spec:
  selector:
    samba-operator.samba.org/service: tshare1
  ports:
  - port: 445
    protocol: TCP
```

This makes it possible to use `//tshare1/My Share` to connect, independent from the IP-address that the Pod currently has:

```console
$ kubectl -n samba-operator-system exec -ti centos -- /bin/bash
[root@centos /]# smbclient -U sambauser '//tshare1/My Share'
Enter SAMBA\sambauser's password: 
Try "help" to get a list of possible commands.
smb: \> 
```

Or, if the consumer of the SmbShare runs in a different namespace, it can use `tshare1.samba-operator-system.svc.cluster.local/My Share`:
```console
$ kubectl exec -ti centos -- /bin/bash
[root@centos /]# smbclient -U sambauser '//tshare1.samba-operator-system.svc.cluster.local/My Share'
Enter SAMBA\sambauser's password: 
Try "help" to get a list of possible commands.
smb: \> 
```

**Comment by phlogistonjohn** [comment]
Partially implemented in PR #52 - however it lacks flexibility. I also have a more advanced form in a branch for my demo at sambaxp - but it is configured in a very hacky way. I plan on doing a design pr soon that will make configuring things like services clearer. Watch this space. :-)

**Comment by phlogistonjohn** [comment]
This is (basically) done. I've finally even wrote documentation for it in #92  - closing this one.
